jib-core/src/main/java/com/google/cloud/tools/jib/builder/steps/AuthenticatePushStep.java

```diff
@@ -21,58 +21,53 @@
 import com.google.cloud.tools.jib.builder.ProgressEventDispatcher;
 import com.google.cloud.tools.jib.builder.TimerEventDispatcher;
 import com.google.cloud.tools.jib.configuration.BuildContext;
-import com.google.cloud.tools.jib.http.Authorization;
-import com.google.cloud.tools.jib.registry.RegistryAuthenticator;
+import com.google.cloud.tools.jib.registry.RegistryClient;
+import com.google.cloud.tools.jib.registry.credentials.CredentialRetrievalException;
 import java.io.IOException;
-import java.util.Optional;
 import java.util.concurrent.Callable;
-import javax.annotation.Nullable;
 
 /**
  * Authenticates push to a target registry using Docker Token Authentication.
  *
  * @see <a
  *     href="https://docs.docker.com/registry/spec/auth/token/">https://docs.docker.com/registry/spec/auth/token/</a>
  */
-class AuthenticatePushStep implements Callable<Optional<Authorization>> {
+class AuthenticatePushStep implements Callable<RegistryClient> {
 
   private static final String DESCRIPTION = "Authenticating push to %s";
 
   private final BuildContext buildContext;
   private final ProgressEventDispatcher.Factory progressEventDispatcherFactory;
-  @Nullable private final Credential registryCredential;
 
   AuthenticatePushStep(
-      BuildContext buildContext,
-      ProgressEventDispatcher.Factory progressEventDispatcherFactory,
-      @Nullable Credential registryCredential) {
+      BuildContext buildContext, ProgressEventDispatcher.Factory progressEventDispatcherFactory) {
     this.buildContext = buildContext;
     this.progressEventDispatcherFactory = progressEventDispatcherFactory;
-    this.registryCredential = registryCredential;
   }
 
   @Override
-  public Optional<Authorization> call() throws IOException, RegistryException {
+  public RegistryClient call() throws CredentialRetrievalException, IOException, RegistryException {
     String registry = buildContext.getTargetImageConfiguration().getImageRegistry();
     try (ProgressEventDispatcher ignored =
             progressEventDispatcherFactory.create("authenticating push to " + registry, 1);
```

Comment:
Does it make sense to make this 2 allocation units now, or is it so quick, this doesn't really matter.

Reply:
I find no problem to have 2 allocations.